rust/src/doc.rs

```diff
@@ -481,6 +476,43 @@ impl Doc {
             .collect())
     }
 
+    pub fn marks_cursor(
```

Comment:
Given that this and `marks_position` are so similar I would suggest unifying the definitions with an enum over the position. Something like this:

```rust
enum Position {
    Cursor(am::Cursor),
    Index(usize),
}
```

pub fn marks_at_position(&self, obj: ObjId, position: Position, heads: Vec<ChangeHash>) { .. }
```

Reply:
Suggestion addressed in 55763e3897fd1273a50e5eb1f1093aaab284b4de